lib/net.js

```diff
@@ -1330,21 +1332,20 @@ Server.prototype.listen = function() {
   var args = new Array(arguments.length);
   for (var i = 0; i < arguments.length; i++)
     args[i] = arguments[i];
-  args = normalizeArgs(args);
+  var [options, cb] = normalizeArgs(args);
 
-  if (args[1]) {
-    this.once('listening', args[1]);
+  if (typeof cb === 'function') {
+    this.once('listening', cb);
   }
 
-  var options = args[0];
-  if (arguments.length === 0 || typeof arguments[0] === 'function') {
+  if (args.length === 0 || typeof args[0] === 'function') {
     // Bind to a random port.
     options.port = 0;
   }
 
   // The third optional argument is the backlog size.
   // When the ip is omitted it can be the second argument.
-  var backlog = toNumber(arguments[1]) || toNumber(arguments[2]);
+  var backlog = toNumber(args[1]) || toNumber(args[2]);
```

Comment:
Do you mean like this?

```js
var backlog = toNumber(args.length > 1 && args[1]) || toNumber(args.length > 2 && args[2]);
```

But is it really necessary / worth it to optimize this function? I don't think it's one of the hot functions in an usual node application, you call listen once during startup and not hundreds of times a second. What is your opinion?